syntax = "proto3";

package social;

import "google/protobuf/timestamp.proto";

service SocialService {
  rpc CreatePost(CreatePostRequest) returns (PostResponse);
  rpc UpdatePost(UpdatePostRequest) returns (PostResponse);
  rpc DeletePost(DeleteRequest) returns (Empty);
  rpc GetPost(GetRequest) returns (PostResponse);
  rpc ListPosts(ListRequest) returns (ListPostsResponse);
}

message Empty {}

message DeleteRequest {
  string id = 1;
  string user_id = 2;
}

message GetRequest {
  string id = 1;
  string user_id = 2;
}

message ListRequest {
  int32 page = 1;
  int32 page_size = 2;
  string user_id = 3;
}

message Post {
  string id = 1;
  string title = 2;
  string description = 3;
  string user_id = 4;
  bool is_private = 5;
  repeated string tags = 6;
  string loyalty_platform = 7;
  google.protobuf.Timestamp created_at = 8;
  google.protobuf.Timestamp updated_at = 9;
}

message CreatePostRequest {
  string title = 1;
  string description = 2;
  string user_id = 3;
  bool is_private = 4;
  repeated string tags = 5;
  string loyalty_platform = 6;
}

message UpdatePostRequest {
  string id = 1;
  string title = 2;
  string description = 3;
  bool is_private = 4;
  repeated string tags = 5;
  string loyalty_platform = 6;
  string user_id = 7;
}

message PostResponse {
  Post post = 1;
}

message ListPostsResponse {
  repeated Post posts = 1;
  int32 total = 2;
}

